### PR TITLE
RCD Overhaul/Major Port

### PR DESCRIPTION
**Pull request by redmoogle**
### Intent of your Pull Request

Ports: https://github.com/tgstation/tgstation/pull/45357
Ports: https://github.com/tgstation/tgstation/pull/45607
Ports: https://github.com/tgstation/tgstation/pull/52018
Ports: https://github.com/tgstation/tgstation/pull/49844

### Why is this good for the game?

Modernize our RCD with some epic features

image
image
image
image

#### Changelog

:cl:  
rscadd: Simple circuits RCD upgrade.
rscadd: Silo link RCD upgrade
rscadd: The RCD's functionality for directional windows has been enabled.
rscadd: Cyborgs can now install RCD upgrades.
rscadd: The RCD's functionality for windowed doors has been enabled.
/:cl:

### CODE REVIEW

**Comment by ToGWtF** [comment]
idk how i feel about RCD silo link. Makes RCD's REALLY powerful

edit: just read, it needs to be researched and linked manually. Good luck convincing AI to let you in

**Comment by redmoogle** [comment]
> 
> 
> idk how i feel about RCD silo link. Makes RCD's REALLY powerful

I'd say testmerge it first and see how much the playerbase abuses it

File: code/modules/research/designs/tool_designs.dm [hunk 1]
```diff
@@ -33,13 +33,43 @@
...
 	build_type = PROTOLATHE
 	materials = list(/datum/material/iron = 5000, /datum/material/glass = 2500, /datum/material/silver = 1500, /datum/material/titanium = 2000)
-	build_path = /obj/item/rcd_upgrade
+	build_path = /obj/item/rcd_upgrade/frames
+	category = list("Tool Designs")
+	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING
+
+/datum/design/rcd_upgrade/simple_circuits
+	name = "RCD simple circuits designs upgrade"
+	desc = "Adds the simple circuits to the RCD."
```
Comment (Hopekz) [inline code/modules/research/designs/tool_designs.dm]:
```suggestion
	desc = "Adds simple circuits to the RCD."
```

File: code/modules/research/designs/tool_designs.dm [hunk 2]
```diff
@@ -33,13 +33,43 @@
...
+	id = "rcd_upgrade_furnishing"
+	build_type = PROTOLATHE
+	materials = list(/datum/material/iron = 5000, /datum/material/glass = 2500, /datum/material/silver = 1500, /datum/material/titanium = 2000)
+	build_path = /obj/item/rcd_upgrade/furnishing
+	category = list("Tool Designs")
+	departmental_flags = DEPARTMENTAL_FLAG_ENGINEERING
+
+/datum/design/rcd_upgrade/silo_link
+	name = "Advanced RCD silo link upgrade"
+	desc = "Adds the silo dirrect link to the RCD."
```
Comment (Hopekz) [inline code/modules/research/designs/tool_designs.dm]:
```suggestion
	desc = "Adds the silo direct link to the RCD."
```

**Comment by maxion12345** [comment]
RCD silo link seems kinda unnecessary. An RCD buff that really isn't required. You can be damn sure someone is gonna end up abusing it to wall off entire sections of hallways before they are stopped, or go on a deconstruction spree. Having to manually feed the RCD atleast limits what you can do on your own, before having to resupply. What about an upgrade to increase the matter amount it can store instead?

 The additions to the building list are pretty nice, but extremely mundane. Tables? Chairs? A chair is literally just a single piece of metal, an RCD isn't needed for that. I'd support changing the table maker to a subset of "fancy tables", such as the gambling table and fancy carpet ones. If your going to bother researching and constructing a furnishing upgrade, make the furnishings good! In addition, we could change the chair to sofas instead. I know that I constantly fuck up the sofas because without a photo to look at, I end up guessing the orientation half the time and going "no, right is left facing this way...". 

Windoors are a nice touch though. Not sure I like the reinforced windoor, because that would require plasteel and deviates from the "Iron/glass" materials the things RCD constructs are made out of. Even the reinforced windows are just rods and glass.

**Comment by ghost** [comment]
@maxion12345 
it's fine, it has to be researched and then manually linked

it's not like you can't reload the RCD with literally just glass or iron anyway

**Comment by maxion12345** [comment]
> @maxion12345
> it's fine, it has to be researched and then manually linked
> 
> it's not like you can't reload the RCD with literally just glass or iron anyway

More worried this will go the route of the RPD and end up being a tool to spam walls and block off large swaths of the station. 
Also imagine clock cult with fuckin ore silo linked RCD, wew. And while having to manually link it would help be a barrier, you could easily just ask the AI and they'll do it no questions asked. 

TL/DR
I feel the RCD was meant to automate tedious common building (walls, windows, airlocks) and not completely replace an engineer having to carry materials on him to do his job. I really don't want to turn it into an all-in-one tool that an engineer can just stash in his pocket. You should at least have to carry materials to refill the damn thing.

**Comment by maxion12345** [comment]
> > @maxion12345
> > it's fine, it has to be researched and then manually linked
> > it's not like you can't reload the RCD with literally just glass or iron anyway
> 
> More worried this will go the route of the RPD and end up being a tool to spam walls and block off large swaths of the station.
> Also imagine clock cult with fuckin ore silo linked RCD, wew. And while having to manually link it would help be a barrier, you could easily just ask the AI and they'll do it no questions asked.
> 
> TL/DR
> I feel the RCD was meant to automate tedious common building (walls, windows, airlocks) and not completely replace an engineer having to carry materials on him to do his job. I really don't want to turn it into an all-in-one tool that an engineer can just stash in his pocket. You should at least have to carry materials to refill the damn thing. A fully upgraded RCD would be a high value item for sure, like something I'd see in a traitor steal objective

**Review by jcatjmeow — APPROVED** [review]
My only concern is this is a new way to reduce the silos' resources quickly. But there are already faster ways to do it via spam printing shit, so eh.

Code's fine

**Comment by JamieD1** [comment]
ID ACCESS ON AIRLOCK DOESNT WORK - PLEASE FIX

**Pull request merged**